File: 소수만들기.cpp
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

int path[100];

vector<int> container;

void abc(int lev, int* vect, int size, int selected) {
    if (selected == 3) {
        int sum = 0;
        int cnt = 0;
        for (int i = 0; i < 3; i++) sum += path[i];
        for (int i = 1; i <= sum; i++) {
            if (sum % i == 0) cnt++;
            if (cnt > 2) return;
        }
        container.push_back(sum);
        return;
    }
    
    for (int i = lev; i < size; i++) {
        path[selected] = vect[i];
        abc(i + 1, vect, size, selected + 1);
        path[selected] = 0;
    }
}

int solution(vector<int> nums) {
    int answer;
    int* arr = nums.data();
    int size = nums.size();
    abc(0, arr, size, 0);
    return container.size();
}

File: 2023-08-31/소수만들기.js
function solution(nums) {
  let answer = [];
  let sumNums = [];
  nums = nums.sort((a, b) => a - b);
  for (let i = 0; i < nums.length - 2; i++) {
    for (let j = i + 1; j < nums.length - 1; j++) {
      for (let k = j + 1; k < nums.length; k++) {
        const sumNum = nums[i] + nums[j] + nums[k];
        sumNums.push(sumNum);
      }
    }
  }
  sumNums.forEach((e) => {
    let a = 0;
    for (let i = 2; i < e; i++) {
      if (!(e % i === 0)) {
        a++;
      }
    }
    if (a === e - 2) {
      answer.push(e);
    }
  });
  return answer.length;
}
console.log(solution([1, 2, 7, 6, 4]));
